src/main/java/uk/gov/pay/products/filters/LoggingMDCRequestFilter.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.pay.products.filters;
+
+import org.slf4j.MDC;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import java.io.IOException;
+import java.util.Optional;
+
+import static uk.gov.pay.logging.LoggingKeys.GATEWAY_ACCOUNT_ID;
+
+public class LoggingMDCRequestFilter implements ContainerRequestFilter {
+    
+    public static final String PAYMENT_EXTERNAL_ID = "payment_external_id";
+    public static final String PRODUCT_EXTERNAL_ID = "product_external_id";
+    public static final String REFERENCE_NUMBER = "reference_number";
+    
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        getPathParameterFromRequest("gatewayAccountId", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(GATEWAY_ACCOUNT_ID, gatewayAccountId));
+
+        getPathParameterFromRequest("paymentExternalId", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(PAYMENT_EXTERNAL_ID, gatewayAccountId));
+
+        getPathParameterFromRequest("productExternalId", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(PRODUCT_EXTERNAL_ID, gatewayAccountId));
+
+        getPathParameterFromRequest("referenceNumber", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(REFERENCE_NUMBER, gatewayAccountId));
```

Comment:
can this variable be renamed to `referenceNumber ` please

Reply:
Done